common/sketch/src/main/java/org/apache/spark/util/sketch/CountMinSketchImpl.java

```diff
@@ -348,4 +374,30 @@ public static CountMinSketchImpl readFrom(InputStream in) throws IOException {
 
     return new CountMinSketchImpl(depth, width, totalCount, hashA, table);
   }
+
+  @Override
+  public void writeExternal(ObjectOutput out) throws IOException {
+    try (ByteArrayOutputStream bos = new ByteArrayOutputStream()) {
+      this.writeTo(bos);
+      byte[] bytes = bos.toByteArray();
+      out.writeObject(bytes);
```

Comment:
Yeah, essentially we can't retrieve the underlying output stream from an `ObjectOutput`. On the other hand, for typical use cases `df.countMinSketch()` this method isn't on performance critical path.